scripts/checksum/index.ts

```diff
@@ -0,0 +1,137 @@
+ * @see {@link https://hg.adblockplus.org/adblockplus/file/tip/addChecksum.py Adblock Plus checksum script}
+ *
+ * @param content The content to hash.
+ * @returns The formatted checksum string.
+ */
+function calculateChecksumMD5(content: string): string {
+    content = normalizeContent(content);
+    const checksum = Base64.stringify(MD5(content));
+
+    return checksum.trim().replace(/=+$/g, '');
+}
+
+/**
+ * Splits a string by lines while preserving line breaks within the original lines.
+ *
+ * @param s The input string to split.
+ * @returns An array of strings where each element is a complete line of text,
+ * including its line break.
+ */
+function splitByLines(s: string): string[] {
+    // Preserve end-of-line characters in the split strings.
+    return s.split(/(?<=\r?\n)/);
+}
+
+/**
+ * Checks if the provided file content contains a checksum tag within its first 200 characters.
+ * This approach is used to exclude parsing checksums from included filters.
+ *
+ * @param file The file content as a string.
+ * @returns `true` if the checksum tag is found, otherwise `false`.
+ */
+function hasChecksum(file: string): boolean {
+    const partOfFile = file.substring(0, 200);
+    const lines = splitByLines(partOfFile);
+
+    return lines.some((line) => line.startsWith(`! ${CHECKSUM_TAG}`));
+}
+
+/**
+ * Creates a tag for filter list metadata.
+ *
+ * @param tagName The name of the tag.
+ * @param value The value of the tag.
+ * @param lineEnding The line ending to use in the created tag.
+ * @returns The created tag in the `! ${tagName}: ${value}` format.
+ */
+function createTag(tagName: string, value: string, lineEnding: string): string {
+    return `! ${tagName}: ${value}${lineEnding}`;
+}
+
+/**
+ * Removes a specified tag from an array of filter content strings.
+ * This function searches for the first occurrence of the specified tag within
+ * the first N lines and removes the entire line containing that tag. If the tag is not
+ * found, the array remains unchanged.
+ *
+ * @param tagName The name of the tag to be removed from the filter content.
+ * @param filterContent An array of strings, each representing a line of filter content.
+ * @returns A new array of filter content with the specified tag removed.
+ * If the tag is not found, the original array is returned unmodified.
+ */
+function removeTag(
+    tagName: string,
+    filterContent: string[],
+): string[] {
+    // Lines of filter metadata to parse
+    const AMOUNT_OF_LINES_TO_PARSE = 50;
+
+    // Make a copy of the array to avoid mutation.
+    const copy = filterContent.slice();
+
+    // Parse only the first 50 lines. Parsing the entire file is unnecessary.
+    const updatedFile = filterContent.slice(
+        0,
+        Math.min(AMOUNT_OF_LINES_TO_PARSE, filterContent.length),
+    );
+
+    const tagIdx = updatedFile.findIndex((line) => line.includes(tagName));
```

Comment:
Are you sure this check is enough? Chances are low, but what if we found a rule like this?

```adblock
##.Title
```

I suggest adding a check for comment marker, e.g

```suggestion
    const regexp = new RegExp(`^[!#]\s*${tagName}:`);
    const tagIdx = updatedFile.findIndex((line) => line.match(regexp));
```

Reply:
These are our filters, and we know their content, so i thing there is not need for changes.